osc-server/src/main/java/org/osc/core/broker/service/tasks/conformance/openstack/sfc/UpdateServiceFunctionChainTask.java

```diff
@@ -56,18 +58,25 @@ public UpdateServiceFunctionChainTask create(ServiceFunctionChain sfc, SecurityG
     }
 
     @Override
-    public String getName() {
-        return String.format("Updating Service Function Chain '%s' for Security Group '%s' under Project '%s'",
-                this.sfc.getName(), this.securityGroup.getName(), this.securityGroup.getProjectName());
-    }
-
-    @Override
     public void executeTransaction(EntityManager em) throws Exception {
         this.sfc = em.find(ServiceFunctionChain.class, this.sfc.getId());
         this.securityGroup = em.find(SecurityGroup.class, this.securityGroup.getId());
-        this.updatedPortPairGroups.size();
-        //TODO: arvind implement this
 
+        try (SdnRedirectionApi sdnApi = this.apiFactory
+                .createNetworkRedirectionApi(this.securityGroup.getVirtualizationConnector())) {
+            NetworkElement updatedChain = sdnApi.updateNetworkElement(
+                    new NetworkElementImpl(this.securityGroup.getNetworkElementId()), this.updatedPortPairGroups);
+            if (!updatedChain.getElementId().equals(this.securityGroup.getNetworkElementId())) {
```

Comment:
Why do we check the updated chain ID? Do we expect a new sfc entity being created on Openstack as a part of updateNetworkElement method?

Reply:
yes. It really depends on the implementation though.  some plugins can consider SFC to be immutable and only way to update is to delete/recreate.

We could get an updated chain id or the existing chain id. Either case, we need to make sure our DB reflects the right values.